pubtools/_pulp/tasks/delete.py

```diff
@@ -0,0 +1,735 @@
+import logging
+import re
+import sys
+import attr
+
+from functools import partial
+
+from more_executors.futures import (
+    f_map,
+    f_flat_map,
+    f_zip,
+    f_return,
+    f_sequence,
+)
+from pushsource import FilePushItem, ModuleMdPushItem, RpmPushItem
+from pubtools.pulplib import (
+    Criteria,
+    Matcher,
+    RpmUnit,
+    ModulemdUnit,
+    FileUnit,
+    ErratumUnit,
+)
+
+from pubtools._pulp.services.pulp import PulpClientService
+from pubtools._pulp.services.collector import CollectorService
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.tasks.common import Publisher
+from pubtools._pulp.arguments import SplitAndExtend
+
+LOG = logging.getLogger("pubtools.pulp")
+
+step = PulpTask.step
+
+MODULEMD_REGEX = re.compile(r"^[-.+\w]+:[-.+\w]+:\d+(:[-.+\w]+){0,2}$")
+
+
+@attr.s
+class ClearedRepo(object):
+    """Represents a single repo where contents were removed."""
+
+    tasks = attr.ib()
+    """The completed Pulp tasks for removing content from this repo."""
+
+    repo = attr.ib()
+    """The repo which where content was cleared."""
+
+    content = attr.ib()
+    """The content that was cleared from the repo"""
+
+
+@attr.s
+class RemoveUnitItem(object):
+    """Represents a unit with destination repos"""
+
+    unit = attr.ib()
+    """A Pulp unit"""
+
+    repos = attr.ib()
+    """The list of repos from where the unit to be removed"""
+
+
+class Delete(PulpClientService, CollectorService, Publisher, PulpTask):
+    """Remove specific content from one or more Pulp repository"""
+
+    def __init__(self, *args, **kwargs):
+        super(Delete, self).__init__(*args, **kwargs)
+        self.to_await = []
+
+    def add_args(self):
+        super(Delete, self).add_args()
+
+        self.add_publisher_args(self.parser)
+
+        self.parser.add_argument(
+            "--repo",
+            help="remove content from these comma-seperated repositories ",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+
+        self.parser.add_argument(
+            "--advisory",
+            help="remove packages and modules in these comma-separated advisories",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+
+        self.parser.add_argument(
+            "--file",
+            help="remove these comma-separated rpm, srpm, modulemd or iso file(s)",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+
+        self.parser.add_argument(
+            "--signing-key",
+            help="remove the content with these signing-keys(s)",
+            type=str,
+            action=SplitAndExtend,
+            split_on=",",
+        )
+
+        self.parser.add_argument(
+            "--no-publish",
```

Comment:
Can this please be implemented as `--skip publish` instead?  That's how it works in clear-repo command - https://release-engineering.github.io/pubtools-pulp/clear-repo.html#example-skipping-publish

In fact if you use a `@step("Publish")` somewhere, you don't have to implement any code explicitly for the skipping, because `@step` does it for you.

Reply:
Added option ```--skip```  and ```@step("Publish")``` exists in the ```Publisher``` class